### PR TITLE
Added a Dockerfile for easy building and running

### PR DESCRIPTION
**Pull request by Tebro**
This PR adds a Docker file that allows the project to be easily built and run. 

My suggestion is setting up a Automated build on Docker Hub that builds the master branch automatically.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+FROM tutum/apache-php
+MAINTAINER FOOBAR
```
Comment (Jalle19) [inline Dockerfile]: Can you add you or me here?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+ENV ALLOW_OVERRIDE=true
+#Git is good for composer
+RUN apt-get update && apt-get install -yq git php5-imagick php5-sqlite php5-json&& rm -rf /var/lib/apt/lists/*
```
Comment (Jalle19) [inline Dockerfile]: Can you split this line so that each command is on separate lines?

**Comment by Jalle19** [comment]
Can you also add "fixes #269" to the commit message?

**Comment by Tebro** [comment]
Done, done and done.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+RUN a2enmod rewrite expires
+
+RUN rm -fr /app
+
+ADD . /src
+
+WORKDIR /src
+RUN composer install
+RUN ./src/protected/yiic createinitialdatabase
+RUN ./src/protected/yiic setpermissions
```
Comment (Jalle19) [inline Dockerfile]: `php ./src/protected/yiic.php migrate --interactive=0` must be run as well.

**Comment by Jalle19** [comment]
@Tebro can you make sure this thing actually works? It doesn't seem to like Vagrant at all.

**Comment by Jalle19** [comment]
Going to merge this now and figure it out later.

**Pull request merged**